packages/zoe/src/contractSupport/zoeHelpers.js

```diff
@@ -3,438 +3,363 @@
-        checkKeys(actual.exit, expected.exit)
-      );
-    },
-    getActiveOffers: handles =>
-      zcf.getOffers(zcf.getOfferStatuses(handles).active),
-    rejectOffer,
-
-    /**
-     * Check whether an update to currentAllocation satisfies
-     * proposal.want. Note that this is half of the offer safety
-     * check; whether the allocation constitutes a refund is not
-     * checked. Allocation is merged with currentAllocation
-     * (allocations' values prevailing if the keywords are the same)
-     * to produce the newAllocation.
-     * @param {OfferHandle} offerHandle
-     * @param {allocation} amountKeywordRecord
-     * @returns {boolean}
-     */
-    satisfies: (offerHandle, allocation) => {
-      const currentAllocation = zcf.getCurrentAllocation(offerHandle);
-      const newAllocation = mergeAllocations(currentAllocation, allocation);
-      const { proposal } = zcf.getOffer(offerHandle);
-      return satisfiesWant(zcf.getAmountMath, proposal, newAllocation);
-    },
+/**
+ * Check if the keywords match expected. Returns a boolean and does
+ * not throw.
+ * @param {ZCFSeat} seat
+ * @param {ExpectedRecord} expected
+ * @returns {boolean}
+ */
+export const checkIfProposal = (seat, expected) => {
+  const actual = seat.getProposal();
+  return (
+    // Check that the "give" keys match expected keys.
+    checkKeys(actual.give, expected.give) &&
+    // Check that the "want" keys match expected keys.
+    checkKeys(actual.want, expected.want) &&
+    // Check that the "exit" key (i.e. "onDemand") matches the expected key.
+    checkKeys(actual.exit, expected.exit)
+  );
+};
 
-    /**
-     * Check whether an update to currentAllocation satisfies offer
-     * safety. Note that this is the equivalent of `satisfiesWant` ||
-     * `satisfiesGive`. Allocation is merged with currentAllocation
-     * (allocations' values prevailing if the keywords are the same)
-     * to produce the newAllocation.
-     * @param {OfferHandle} offerHandle
-     * @param {AmountKeywordRecord} allocation
-     * @returns {boolean}
-     */
-    isOfferSafe: (offerHandle, allocation) => {
-      const currentAllocation = zcf.getCurrentAllocation(offerHandle);
-      const newAllocation = mergeAllocations(currentAllocation, allocation);
-      const { proposal } = zcf.getOffer(offerHandle);
-      return isOfferSafe(zcf.getAmountMath, proposal, newAllocation);
-    },
+/**
+ * Check whether an update to currentAllocation satisfies
+ * proposal.want. Note that this is half of the offer safety
+ * check; whether the allocation constitutes a refund is not
+ * checked. Allocation is merged with currentAllocation
+ * (allocations' values prevailing if the keywords are the same)
+ * to produce the newAllocation.
+ * @param {ContractFacet} zcf
+ * @param {ZCFSeat} seat
+ * @param {Allocation} allocation
+ * @returns {boolean}
+ */
+export const satisfies = (zcf, seat, allocation) => {
```

Comment:
The `allocation` parameter to `satisfies` is not an Allocation. It is an AmountKeywordRecord representing an update to an allocation.

```suggestion
 * @param {AmountKeywordRecord} allocationUpdate
 * @returns {boolean}
 */
export const satisfies = (zcf, seat, allocationUpdate) => {
```

The comment also needs adjusting.